core/trace.c

```diff
@@ -37,12 +40,16 @@ THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
 
 // FIXME: This is a hack to allow trace.c to get the threaded support API
 //  even though we are using a unthreaded runtime
-#define _LF_TRACE
+// #define _LF_TRACE
```

Comment:
This is causing some of the CI failures with unthreaded programs with tracing enabled. That commented out define is to enable tracing with the unthreaded runtime. A solution would be to make unthreaded version of the tracing,  I think this is the most natural approach, it would work with e.g. the Arduino target as well. I think it will be surprising for our users that even if you specify `threading: false` we will link with a threading library if you specify `tracing: true`.

Reply:
I have simplified the tracing code so it doesn't need threads.  There will be a performance hit when trace buffers have to be flushed to files, but that should occur rarely, and for most useful trace files, not until shutdown.